File: app/src/main/java/com/giosis/util/qdrive/settings/ScanSettingActivity.java
package com.giosis.util.qdrive.settings;

import android.app.Activity;
import android.content.Context;
import android.content.SharedPreferences;
import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;
import android.view.View;
import android.widget.Button;
import android.widget.FrameLayout;
import android.widget.TextView;

import com.giosis.util.qdrive.singapore.R;

public class ScanSettingActivity extends AppCompatActivity {
    String TAG = "ScanSettingActivity";


    FrameLayout layout_top_back;
    TextView text_top_title;

    Button btn_scan_setting_vibration_on;
    Button btn_scan_setting_vibration_off;


    Context context;
    String vibrationString;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_scan_setting);

        layout_top_back = findViewById(R.id.layout_top_back);
        text_top_title = findViewById(R.id.text_top_title);

        btn_scan_setting_vibration_on = findViewById(R.id.btn_scan_setting_vibration_on);
        btn_scan_setting_vibration_off = findViewById(R.id.btn_scan_setting_vibration_off);


        //
        context = getApplicationContext();

        text_top_title.setText(R.string.text_title_scan_setting);


        layout_top_back.setOnClickListener(clickListener);
        btn_scan_setting_vibration_on.setOnClickListener(clickListener);
        btn_scan_setting_vibration_off.setOnClickListener(clickListener);

        //
        SharedPreferences sharedPreferences = getSharedPreferences("PREF_SCAN_SETTING", Activity.MODE_PRIVATE);
        vibrationString = sharedPreferences.getString("vibration", "0");

        if (vibrationString.equals("0")) {
            // MainActivity 에서 초기화, 혹시 안됐을 경우..

            vibrationString = "OFF";

            SharedPreferences.Editor editor = sharedPreferences.edit();
            editor.putString("vibration", vibrationString);
            editor.apply();

            btn_scan_setting_vibration_on.setBackgroundResource(R.drawable.custom_button_normal_30);
            btn_scan_setting_vibration_on.setTextColor(context.getResources().getColor(R.color.color_4fb648));
            btn_scan_setting_vibration_off.setBackgroundResource(R.drawable.custom_button_pressed_30);
            btn_scan_setting_vibration_off.setTextColor(context.getResources().getColor(R.color.white));
        } else {

            if (vibrationString.equals("ON")) {

                btn_scan_setting_vibration_on.setBackgroundResource(R.drawable.custom_button_pressed_30);
                btn_scan_setting_vibration_on.setTextColor(context.getResources().getColor(R.color.white));
                btn_scan_setting_vibration_off.setBackgroundResource(R.drawable.custom_button_normal_30);
                btn_scan_setting_vibration_off.setTextColor(context.getResources().getColor(R.color.color_4fb648));
            } else if (vibrationString.equals("OFF")) {

                btn_scan_setting_vibration_on.setBackgroundResource(R.drawable.custom_button_normal_30);
                btn_scan_setting_vibration_on.setTextColor(context.getResources().getColor(R.color.color_4fb648));
                btn_scan_setting_vibration_off.setBackgroundResource(R.drawable.custom_button_pressed_30);
                btn_scan_setting_vibration_off.setTextColor(context.getResources().getColor(R.color.white));
            }
        }
    }


    View.OnClickListener clickListener = new View.OnClickListener() {
        @Override
        public void onClick(View v) {

            switch (v.getId()) {
                case R.id.layout_top_back: {

                    finish();
                }
                break;

                case R.id.btn_scan_setting_vibration_on: {

                    if (vibrationString.equals("OFF")) {        // OFF > ON

                        vibrationString = "ON";

                        btn_scan_setting_vibration_on.setBackgroundResource(R.drawable.custom_button_pressed_30);
                        btn_scan_setting_vibration_on.setTextColor(context.getResources().getColor(R.color.white));
                        btn_scan_setting_vibration_off.setBackgroundResource(R.drawable.custom_button_normal_30);
                        btn_scan_setting_vibration_off.setTextColor(context.getResources().getColor(R.color.color_4fb648));

                        SharedPreferences sharedPreferences = getSharedPreferences("PREF_SCAN_SETTING", Activity.MODE_PRIVATE);
                        SharedPreferences.Editor editor = sharedPreferences.edit();
                        editor.putString("vibration", vibrationString);
                        editor.apply();
                    }
                }
                break;
                case R.id.btn_scan_setting_vibration_off: {

                    if (vibrationString.equals("ON")) {         // ON > OFF

                        vibrationString = "OFF";

                        btn_scan_setting_vibration_on.setBackgroundResource(R.drawable.custom_button_normal_30);
                        btn_scan_setting_vibration_on.setTextColor(context.getResources().getColor(R.color.color_4fb648));
                        btn_scan_setting_vibration_off.setBackgroundResource(R.drawable.custom_button_pressed_30);
                        btn_scan_setting_vibration_off.setTextColor(context.getResources().getColor(R.color.white));

                        SharedPreferences sharedPreferences = getSharedPreferences("PREF_SCAN_SETTING", Activity.MODE_PRIVATE);
                        SharedPreferences.Editor editor = sharedPreferences.edit();
                        editor.putString("vibration", vibrationString);
                        editor.apply();
                    }
                }
                break;
            }
        }
    };
}

File: appMY/src/main/java/com/giosis/util/qdrive/settings/ScanSettingActivity.kt
package com.giosis.util.qdrive.settings

import android.os.Bundle
import android.util.Log
import com.giosis.util.qdrive.international.MyApplication
import com.giosis.util.qdrive.international.R
import com.giosis.util.qdrive.util.ui.CommonActivity
import kotlinx.android.synthetic.main.activity_scan_setting.*
import kotlinx.android.synthetic.main.top_title.*

class ScanSettingActivity : CommonActivity() {

    val tag = "ScanSettingActivity"
    private val context = MyApplication.getContext()

    lateinit var vibration: String


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_scan_setting)


        text_top_title.text = context.resources.getString(R.string.text_title_scan_setting)


        vibration = MyApplication.preferences.scanVibration
        Log.e(tag, "Vibration : $vibration")


        if (vibration == "OFF") {

            btn_scan_setting_vibration_on.setBackgroundResource(R.drawable.border_radius_30_4fb648_ffffff)
            btn_scan_setting_vibration_on.setTextColor(context.resources.getColor(R.color.color_4fb648))

            btn_scan_setting_vibration_off.setBackgroundResource(R.drawable.bg_radius_30_4fb648)
            btn_scan_setting_vibration_off.setTextColor(context.resources.getColor(R.color.white))
        } else if (vibration == "ON") {

            btn_scan_setting_vibration_on.setBackgroundResource(R.drawable.bg_radius_30_4fb648)
            btn_scan_setting_vibration_on.setTextColor(context.resources.getColor(R.color.white))

            btn_scan_setting_vibration_off.setBackgroundResource(R.drawable.border_radius_30_4fb648_ffffff)
            btn_scan_setting_vibration_off.setTextColor(context.resources.getColor(R.color.color_4fb648))
        }


        layout_top_back.setOnClickListener {

            finish()
        }

        btn_scan_setting_vibration_on.setOnClickListener {

            if (vibration == "OFF") {        // OFF > ON

                vibration = "ON"

                btn_scan_setting_vibration_on.setBackgroundResource(R.drawable.bg_radius_30_4fb648)
                btn_scan_setting_vibration_on.setTextColor(context.resources.getColor(R.color.white))

                btn_scan_setting_vibration_off.setBackgroundResource(R.drawable.border_radius_30_4fb648_ffffff)
                btn_scan_setting_vibration_off.setTextColor(context.resources.getColor(R.color.color_4fb648))

                MyApplication.preferences.scanVibration = vibration
            }
        }

        btn_scan_setting_vibration_off.setOnClickListener {

            if (vibration == "ON") {        // ON > OFF

                vibration = "OFF"

                btn_scan_setting_vibration_on.setBackgroundResource(R.drawable.border_radius_30_4fb648_ffffff)
                btn_scan_setting_vibration_on.setTextColor(context.resources.getColor(R.color.color_4fb648))

                btn_scan_setting_vibration_off.setBackgroundResource(R.drawable.bg_radius_30_4fb648)
                btn_scan_setting_vibration_off.setTextColor(context.resources.getColor(R.color.white))

                MyApplication.preferences.scanVibration = vibration
            }
        }
    }
}